foundry/pixi/containers/canvasLayers/placeablesLayers/templateLayer.d.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * This Canvas Layer provides a container for MeasuredTemplate objects.
+ * @see {@link MeasuredTemplate}
+ */
+declare class TemplateLayer extends PlaceablesLayer<MeasuredTemplate> {
+  /** @override */
+  static get layerOptions(): PlaceablesLayer.LayerOptions;
+
+  /** @override */
+  activate(): this;
+
+  /** @override */
+  deactivate(): this;
+
+  /**
+   * Register game settings used by the TemplatesLayer
+   */
+  static registerSettings(): void;
+
+  /** @override */
+  protected _onDragLeftStart(event: PIXI.InteractionEvent): void;
+
+  /** @override */
+  protected _onDragLeftMove(event: PIXI.InteractionEvent): void;
+
+  /** @override
+   * @remarks Returns `Promise<MeasuredTemplate> | undefined`
```

Comment:
Does just setting this as a return type not work?

Reply:
No, because the base class can return an array in the promise according to the current typings.

Reply:
Actually, it seems more complicated than that. But this one does seem to return a completely different type than the base class, so.